flink/v1.17/flink/src/test/java/org/apache/iceberg/flink/source/reader/ReaderFunctionTestBase.java

```diff
@@ -49,19 +52,18 @@ public static Object[][] parameters() {
     };
   }
 
-  @ClassRule public static final TemporaryFolder TEMPORARY_FOLDER = new TemporaryFolder();
+  @TempDir protected Path temporaryFolder;
 
   protected abstract ReaderFunction<T> readerFunction();
 
   protected abstract void assertRecords(List<Record> expected, List<T> actual, Schema schema);
 
-  private final FileFormat fileFormat;
-  private final GenericAppenderFactory appenderFactory;
+  @Parameter(index = 0)
+  private FileFormat fileFormat;
 
-  public ReaderFunctionTestBase(FileFormat fileFormat) {
-    this.fileFormat = fileFormat;
-    this.appenderFactory = new GenericAppenderFactory(TestFixtures.SCHEMA);
-  }
+  @Parameter(index = 1)
```

Comment:
it seems weird that this is a parameter that isn't actually used in the list of parameters in this class. I think this also needs to be fixed for Flink 1.19

Reply:
Yes, the parameterization isn't needed here. Fix all the versions.